ftw/news/vocabularies.py

```diff
@@ -0,0 +1,41 @@
+from hashlib import md5
+from plone.i18n.normalizer.interfaces import IIDNormalizer
+from Products.CMFCore.utils import getToolByName
+from zope.component import queryUtility
+from zope.interface import directlyProvides
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+from zope.site.hooks import getSite
+
+
+class SubjectVocabulary(SimpleVocabulary):
+
+    def __init__(self, terms, *interfaces):
+        terms = self.get_terms()
+        super(SubjectVocabulary, self).__init__(terms, *interfaces)
+
+    def search(self, query_string):
+        return [v for v in self if query_string.lower() in v.value.lower()]
+
+    def get_terms(self):
+        site = getSite()
+        catalog = getToolByName(site, 'portal_catalog', None)
+        normalizer = queryUtility(IIDNormalizer)
+        terms = []
+
+        for term in catalog.uniqueValuesFor('Subject'):
+            # Just the normalized term is not enough for an unique token.
+            # The unique values are case sensitive. After normalizing the
+            # term, we get id's in lower-case. That means, we lose the
+            # unique value after normalizing if we get values like: "Bond"
+            # and "bonD". Adding an md5 hash to the normalized term, we
+            # prevent double tokens and the token is still readable.
+            token = '{0}-{1}'.format(md5(term).hexdigest(),
+                                     normalizer.normalize(term))
+            terms.append(SimpleTerm(value=term,
```

Comment:
The argument types to `SimpleTerm` are not consistent:
`value` and `token` may be unicode or bytestring, but the `title` is always unicode.

Shouldn't everything be unicode?

I also assume that `term` is a bytestring, otherwise the md5 generation would fail (it does not support unicode).
Did you test umlauts as subjects?